components/raftstore/src/store/entry_storage.rs

```diff
@@ -248,8 +253,12 @@ impl EntryCache {
                 }
                 break;
             }
-            let (_, dangle_size) = cached_entries.take_entries();
-            mem_size_change -= dangle_size as i64;
+            // keep the applying entries if not all of them are persisted.
```

Comment:
If not, where would be a problem?

Reply:
If not all of the entries are persisted and these entries are evicted from both the entry_cache and the cached_entries, then ApplyFsm may not be able to fetch the dangling entries from raft-engine because they are not persisted, thus it will cause tikv panic.

Reply:
You should update above comment 
```
// example, if entries [1, 10), [10, 20), [20, 30) are sent to apply threads and
// `compact_to(15)` is called, only [20, 30) will still be kept in cache.
```

Now `[10,20)` would be kept as well, is that right?

Reply:
No. [10, 20] will be remove from entry_cache, but the entries in CachedEntries(refered by a ApplyMsg) will not be removed.